src/cracker.c

```diff
@@ -477,11 +481,18 @@ static char *crk_loaded_counts(void)
 	if (crk_db->password_count == 1)
 		return "Remaining 1 hash";
 
-	sprintf(s_loaded_counts,
-		"Remaining %d hashes with %s different salts",
-		crk_db->password_count,
-		crk_db->salt_count > 1 ?
-		jtr_itoa(crk_db->salt_count, nbuf, 24, 10) : "no");
+	int p = sprintf(s_loaded_counts,
+	                "Remaining %d hashes with %s different salts",
+	                crk_db->password_count,
+	                crk_db->salt_count > 1 ?
+	                jtr_itoa(crk_db->salt_count, nbuf, 24, 10) : "no");
+
+	int b = (10 * crk_db->password_count + (crk_db->salt_count / 2)) /
+		crk_db->salt_count;
+
+	if (crk_db->salt_count > 1 && b > 10)
+		sprintf(s_loaded_counts + p, " (%d.%dx same-salt boost)",
+		        b / 10, b % 10);
```

Comment:
`sprintf` isn't supposed to fail, but formally it's defined to return a negative value on error. If the first `sprintf` call here fails, then the second one will use an out-of-bounds pointer. I think we should avoid this coding pattern, perhaps by checking that `p > 0` before attempting the second `sprintf`

Reply:
Good point, I'll look into that